app/scripts/services/validateRegistrant.js

```diff
@@ -4,52 +4,36 @@ angular.module('confRegistrationWebApp')
   .service('validateRegistrant', function validateRegistrant() {
 
     var blockVisibleRuleCheck = function (block, registrant) {
-      var returnValueAND = true;
-      var returnValueOR = false;
       var answers = registrant.answers;
-      var ruleOperand = block.content && block.content !== '' && !angular.isUndefined(block.content.ruleoperand) ? block.content.ruleoperand : 'AND';
+      var ruleOperand = block.content && block.content.ruleoperand ? block.content.ruleoperand : 'AND';
+      var ruleArray = [];
 
-      if (ruleOperand === 'AND') {
-        angular.forEach(block.rules, function (rule) {
-          var answer = _.find(answers, { blockId: rule.parentBlockId });
```

Comment:
How come you went back to a for loop instead of a foreach loop? You aren't using the index.